tidy3d/components/data/monitor_data.py

```diff
@@ -1147,19 +1160,20 @@ def _find_ordering_one_freq(
         if num_modes_to_sort <= 1:
             return pairs, complex_amps
 
-        # Compute an overlap matrix for modes chosen for sorting
-        amps_reduced = np.zeros((num_modes_to_sort, num_modes_to_sort), dtype=np.complex128)
-
         # Extract all modes of interest from template data
         data_template_reduced = self._isel(mode_index=modes_to_sort)
 
-        for i, mode_index in enumerate(modes_to_sort):
+        amps_reduced = data_template_reduced.outer_dot(
+            data_to_sort._isel(mode_index=modes_to_sort)
+        ).to_numpy()[0, :, :]
+
+        for i, _mode_index in enumerate(modes_to_sort):
             # Get one mode from data_to_sort
 
-            one_mode = data_to_sort._isel(mode_index=[mode_index])
+            # one_mode = data_to_sort._isel(mode_index=[mode_index])
 
             # Project to all modes of interest from data_template
-            amps_reduced[:, i] = data_template_reduced.dot(one_mode).data.ravel()
+            # amps_reduced[:, i] = data_template_reduced.dot(one_mode).data.ravel()
             if self.monitor.store_fields_direction == "-":
                 amps_reduced[:, i] *= -1
```

Comment:
Seems like the whole loop can be removed then since the `-1` factor can be applied to all of `amps_reduced`.